src/main.py

```diff
@@ -34,29 +25,47 @@ def run_simulation(
     no_of_generations: int,
     report_every_n: int,
     report_show: bool,
-):
+) -> None:
+    """
+    Run simulation with given parameters
+    :param G: Graph representing the city
+    :param all_stops: All stops in the city (vertices of the graph)
+    :param best_paths: Shortest paths between all stops
+    :param no_of_generations: Number of generations to simulate
+    :param report_every_n: Report every n-th generation
+    :param report_show: if <True> run plt.show(), otherwise plt.savefig(...)
+    """
     line_mutator = LineMutator(G, all_stops, best_paths)
     genotype_mutator = GenotypeMutator(G, best_paths)
     genotype_crosser = GenotypeCrosser(G, best_paths)
     sanitizer = BasicSanitizer(best_paths)
     params = NewGenerationRandomParams(
-        CHANCE_CREATE_LINE,
-        CHANCE_CYCLE,
-        CHANCE_ERASE_LINE,
-        CHANCE_INVERT,
-        CHANCE_MERGE,
-        CHANCE_MERGE_SPECIMEN,
-        CHANCE_ROT_CYCLE,
-        CHANCE_ROT_RIGHT,
-        CHANCE_SPLIT,
+        chance_create_line=0.1,
+        chance_cycle=0.1,
+        chance_erase_line=0.1,
+        chance_invert=0.5,
+        chance_merge=0.25,
+        chance_merge_specimen=0.5,
+        chance_rot_cycle=0.5,
+        chance_rot_right=0.5,
+        chance_split=0.75,
+        chance_erase_stop=0.1,
+        chance_add_stop=0.3,
+        chance_add_stop_mix=0.1,
+        chance_replace_stops=0.1,
+        chance_replace_stops_proximity=0.1,
+        chance_merge_mix=0.5,  # brak wplywu
+        cycle_stops_shift=0.5,  # brak wplywu
+        chance_cycle_stops_shift=0.1,
+        chance_line_based_merge=0.1,
```

Comment:
I assume these are the results of grid search optimization?

Reply:
cc @MrJake222